src/main/java/com/google/research/bleth/simulator/AbstractSimulation.java

```diff
@@ -68,15 +74,21 @@ void updateObserversAwaknessState() {
      */
     void beaconsToObservers() {
         for (Beacon beacon : beacons) {
+            boolean observed = false; // for stats
```

Comment:
Prefer to minimize comments; they are prone to rot (as it's easy to change code and forget to change the comment).
Instead, write the code in a way it will be clear even without the comment.
In this particular case, the comment just isn't needed - I can easily understand what `observed` *is*, and reading further down shows me it's used for stats just by its impact on them.